nntrainer/tensor/tensor.cpp

```diff
@@ -3440,18 +3439,20 @@ Tensor &Tensor::erf(Tensor &out) const {
   return out;
 }
 
-void Tensor::sin_transform(Tensor &out, float alpha) {
+void Tensor::sin(Tensor &out, float alpha) {
+  NNTR_THROW_IF(!contiguous, std::invalid_argument)
+    << getName() << " is not contiguous, cannot calculate sin.";
```

Comment:
shouldn't it also check for if the tensor size is equal? 
```
if (size() != out.size())
 throw std::invalid_argument("Size of tensor must match");
```

Reply:
or maybe we can compare TensorDim

Reply:
Makes sense